graphdatascience/query_runner/cypher_graph_constructor.py

```diff
@@ -46,23 +77,25 @@ def run(self, node_dfs: List[DataFrame], relationship_dfs: List[DataFrame]) -> N
                 "edition graph construction (slower)"
             )
 
-        if len(node_dfs) > 1:
-            raise ValueError("The GDS Community edition graph construction supports only a single node dataframe")
-
-        if len(relationship_dfs) > 1:
-            raise ValueError("The GDS Community edition graph construction supports at most one relationship dataframe")
-
-        node_df = node_dfs[0]
-        rel_df = relationship_dfs[0]
-
         # Cypher aggregation supports concurrency since 2.3.0
         if self._server_version >= ServerVersion(2, 3, 0):
             self.CypherAggregationRunner(
                 self._query_runner, self._graph_name, self._concurrency, self._undirected_relationship_types
-            ).run(node_df, rel_df)
+            ).run(node_dfs, relationship_dfs)
         else:
             assert not self._undirected_relationship_types, "This should have been raised earlier."
 
+            if len(node_dfs) > 1:
+                raise ValueError("The GDS Community edition graph construction supports only a single node dataframe")
+
+            if len(relationship_dfs) > 1:
+                raise ValueError(
+                    "The GDS Community edition graph construction supports at most one relationship dataframe"
```

Comment:
We should indicate in the error message that this limitation is tied to GDS versions prior to 2.3